plutus-core/plutus-core/src/PlutusCore/ParserCommon.hs

```diff
@@ -29,24 +27,11 @@ import           Data.Proxy                 (Proxy (Proxy))
 newtype ParserState = ParserState { identifiers :: M.Map T.Text PLC.Unique }
     deriving (Show)
 
-data ParseError = UnknownBuiltinType T.Text
-                | BuiltinTypeNotAStar T.Text
-                | InvalidConstant T.Text T.Text
-                deriving (Eq, Ord, Show)
-
-instance HasErrorCode ParseError where
-      errorCode UnknownBuiltinType {}  = ErrorCode 5
-      errorCode BuiltinTypeNotAStar {} = ErrorCode 52
-      errorCode InvalidConstant {}     = ErrorCode 4
-
-type Error = Parsec.ParseError Char ParseError
+type Parser ann = ParsecT (PLC.ParseError SourcePos) T.Text (StateT ParserState PLC.Quote)
+instance (Stream s, PLC.MonadQuote m) => PLC.MonadQuote (ParsecT e s m)
 
-instance ShowErrorComponent ParseError where
-    showErrorComponent (UnknownBuiltinType ty) = "Unknown built-in type: " ++ T.unpack ty
-    showErrorComponent (BuiltinTypeNotAStar ty) =
-        "Expected a type of kind star (to later parse a constant), but got: " ++ T.unpack ty
-    showErrorComponent (InvalidConstant ty con) =
-        "Invalid constant: " ++ T.unpack con ++ " of type " ++ T.unpack ty
+instance Pretty SourcePos where
```

Comment:
Hmm, not sure if this belongs here. We need it for the `Pretty` instance for errors, I guess?